src/c_api.cpp

```diff
@@ -1315,7 +1320,7 @@ int LGBM_BoosterPredictForCSR(BoosterHandle handle,
   Booster* ref_booster = reinterpret_cast<Booster*>(handle);
   auto get_row_fun = RowFunctionFromCSR(indptr, indptr_type, indices, data, data_type, nindptr, nelem);
   int nrow = static_cast<int>(nindptr - 1);
-  ref_booster->Predict(num_iteration, predict_type, nrow, get_row_fun,
+  ref_booster->Predict(num_iteration, predict_type, nrow, static_cast<int32_t>(num_col), get_row_fun,
```

Comment:
Is it safe? Maybe simply change the original type, as it was not used before?

Reply:
change the original will cause lots of API changes. 
I think we actually don't support too many columns.
I will add a check for the num_col in CSR format, to ensure its value is smaller than int32_max.